src/sage/schemes/elliptic_curves/ell_point.py

```diff
@@ -153,14 +151,44 @@
     PariError = ()
 
 
-class EllipticCurvePoint(SchemeMorphism_point_projective_ring):
+class EllipticCurvePoint(AdditiveGroupElement,
+                         SchemeMorphism_point_projective_ring):
     """
     A point on an elliptic curve.
     """
-    pass
+    def curve(self):
+        """
+        Return the curve that this point is on.
+
+        This is a synonym for :meth:`scheme`.
 
+        EXAMPLES::
 
-class EllipticCurvePoint_field(SchemeMorphism_point_abelian_variety_field):
+            sage: E = EllipticCurve('389a')
+            sage: P = E([-1, 1])
+            sage: P.curve()
+            Elliptic Curve defined by y^2 + y = x^3 + x^2 - 2*x over Rational Field
+
+            sage: E = EllipticCurve(QQ, [1, 1])
+            sage: P = E(0, 1)
+            sage: P.scheme()
+            Elliptic Curve defined by y^2 = x^3 + x + 1 over Rational Field
+            sage: P.scheme() == P.curve()
+            True
+            sage: x = polygen(ZZ, 'x')
+            sage: K.<a> = NumberField(x^2 - 3,'a')                                      # needs sage.rings.number_field
+            sage: P = E.base_extend(K)(1, a)                                            # needs sage.rings.number_field
+            sage: P.scheme()                                                            # needs sage.rings.number_field
+            Elliptic Curve defined by y^2 = x^3 + x + 1 over Number Field in a with defining polynomial x^2 - 3
+        """
+        return self.scheme()
+
+
+# While the class EllipticCurvePoint itself currently provides little
+# functionality, inheritance from EllipticCurvePoint is checked in the
+# __contains__() and __call__() methods of EllipticCurve_generic.
+class EllipticCurvePoint_field(EllipticCurvePoint,
```

Comment:
I guess that the comment in lines 187-190 should go to the class definition ` EllipticCurvePoint`. Or you may just delete it since it has little content.

Reply:
The comment was meant to clarify the new inheritance, but I agree it can be removed.

Reply:
Thanks.